src/credentials_rotators/npm/lambda_functions/user_access_token_rotator.py

```diff
@@ -0,0 +1,103 @@
+import json
+
+from npm_credentials_rotator import NPMCredentialsRotator
+from npm_utils import (create_access_token, delete_access_token, get_user_info_using_access_token)
+from secrets_config_utils import (get_access_token_secrets_configs, get_secret_key)
+from secrets_manager_utils import get_secret_value
+
+
+class UserAccessTokenRotator(NPMCredentialsRotator):
+    """
+    Handles the rotation logic specific to rotating the NPM user login password secret
+    """
+
+    def __init__(self, arn, token, step):
+        super().__init__(arn, token, step)
+        self.secret_config = self.get_access_token_secret_config()
+
+    def create_secret(self):
+        """Create the secret
+        This method first checks for the existence of a secret for the passed in request token. If one does not exist,
+         it will create a new access token for the NPM user and put it with the passed in client request token.
+        Raises:
+            ResourceNotFoundException: If the secret with the specified arn and stage does not exist
+        """
+        # Make sure the current secret exists
+        self.check_secret_exists()
```

Comment:
what happens here if the secret doesn't exist? does this just fail with the error raised in this call? excuse my python ignorance

Reply:
Yes. It raises a `ResourceNotFoundException` and fails